tensorboard/webapp/theme/_tb_theme.template.scss

```diff
@@ -52,3 +52,12 @@ $tb-background: map-get($tb-theme, background);
 
 // Include all theme-styles for the components based on the current theme.
 @include angular-material-theme($tb-theme);
+
+.tb-text-input {
+  -webkit-appearance: none;
+  box-sizing: content-box;
+  font: inherit;
+  border: none;
+  outline: none;
+  padding: 0;
+}
```

Comment:
Instead of creating a CSS based components, just create an Angular widget.

Reply:
Creating a widget wrapper around `<input>` seems to add unnecessary boilerplate, imo.  Every attribute like `autocomplete` or event binding like `(input)` now requires extra @Input's and @Output's on this new widget.  Also, every module that uses it must now import another NgModule.

What do we gain from a TBTextInputWidget, from this extra boilerplate?

Reply:
If it has very consistent UX that we can supply with it, I think it is very well worth it. Kind of like mat-input; for me, I gain almost nothing by using it yet for the consistent UX, we use it. 

CSS based widgetization is creating a new pattern unnecessarily.

Reply:
Switched to the custom widget approach, PTAL